app/assets/javascripts/modules/toggle-attribute.js

```diff
@@ -2,19 +2,19 @@ window.GOVUK = window.GOVUK || {}
 window.GOVUK.Modules = window.GOVUK.Modules || {};
 
 (function (Modules) {
-  'use strict'
-
-  Modules.ToggleAttribute = function () {
-    this.start = function ($element) {
-      $element.find('[data-toggle-attribute]').click(function (event) {
-        var clicked = event.target
-        var toggleAttribute = clicked.getAttribute('data-toggle-attribute')
-        var current = clicked.getAttribute(toggleAttribute)
-        var closedText = clicked.getAttribute('data-when-closed-text')
-        var openText = clicked.getAttribute('data-when-open-text')
-
-        clicked.setAttribute(toggleAttribute, current === closedText ? openText : closedText)
-      })
-    }
+  function ToggleAttribute ($module) {
+    this.$module = $module
+  }
+  ToggleAttribute.prototype.init = function ($element) {
+    $element = this.$module.querySelector('[data-toggle-attribute]')
+    $element.addEventListener('click', function (event) {
```

Comment:
Based on @kevindew's explanation [on this PR](https://github.com/alphagov/collections/pull/2493#pullrequestreview-716050444), it might be more idiomatic to do something like in order to keep the setup code in the constructor separate from the `init` code...

```
'use strict'

Modules.ToggleAttribute = function (element) {
  this.toggleAttribute = element.querySelector('[data-toggle-attribute]')
}

Modules.ToggleAttribute.prototype.init = function () {
  this.toggleAttribute.addEventListener('click', function (event) {
    // ...
  })
}
```

Reply:
Done :) thanks